vuestorefront/models/product_template.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2023 ODOOGAP/PROMPTEQUATION LDA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+
+from odoo import api, fields, models
+from odoo.osv import expression
+from odoo.tools.sql import column_exists, create_column
+
+from odoo.addons.http_routing.models.ir_http import slug
+
+from ..utils import slugify_with_sfx
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _auto_init(self):
+        """Create/set slug via SQL to save on memory usage."""
+        cr = self.env.cr
+        if not column_exists(cr, "product_template", "slug"):
+            create_column(cr, "product_template", "slug", "varchar")
+            cr.execute("SELECT id, name FROM product_template WHERE name IS NOT NULL")
+            for res in cr.fetchall():
+                id_, name = res
+                cr.execute(
+                    "UPDATE product_template SET slug = %s WHERE id = %s",
+                    (slugify_with_sfx(name, f"-{id_}"), id_),
+                )
+        return super()._auto_init()
+
+    def _get_public_categ_slug(self, category_ids, category):
+        category_ids.append(category.id)
+
+        if category.parent_id:
+            category_ids = self._get_public_categ_slug(category_ids, category.parent_id)
+
+        return category_ids
+
+    @api.depends("name")
+    def _compute_slug(self):
+        for product in self:
+            # To make sure we don't assign ID before record is created.
+            if not isinstance(product.id, models.NewId):
+                product.slug = slugify_with_sfx(product.name, f"-{product.id}")
+
+    @api.depends("product_variant_ids")
+    def _compute_variant_attribute_value_ids(self):
+        """Compute attribute values.
+
+        This method computes a list of attribute values from variants of
+        published products.
+        This will ensure that the available attribute values on the website
+        filtering will return results.
+        By default, Odoo only shows attributes that will return results but
+        doesn't consider that a particular
+        attribute value may not have a variant.
+        """
+        for product in self:
+            variants = product.product_variant_ids
+            attribute_values = variants.mapped(
+                "product_template_attribute_value_ids"
+            ).mapped("product_attribute_value_id")
+            product.variant_attribute_value_ids = [(6, 0, attribute_values.ids)]
+
+    variant_attribute_value_ids = fields.Many2many(
```

Comment:
Field declarations should be before compute methods [[docs](https://github.com/versada/versada-docs/blob/master/docs/dev/dev-guide.rst#35models)]

Reply:
Yeah, leftover

Reply:
But I think I will remove this entirely as its kind of weird computation, which can also have big impact on performance and might not be very relevant.